### DISCUSSION TITLE
Can't access to nodejs/Django port servers with HTTPS

### DISCUSSION THREAD

**Discussion by gelintonx** [coder/code-server#1986] [Q&A] [answered]
- Web Browser: Safari and Chrome
- Local OS: macOS Catalina 10.15.5
- Remote OS: Ubuntu 18.04
- Remote Architecture: x86
- `code-server --version`: 3.4.1

Hi, I have got a problem when I activate HTTPS with Let´s Encrypt. I can't access server ports. Per Example I am coding with Django and I need to visit the site that I am developing so I type my domain.com:8000 and it's not working. What should I have to do? If I deactivate Let´s Encrypt I can't code with iPad but I can access to the sites that I am developing. Thanks in advance.

**Comment by code-asher** [comment]
Hey @gelintonx I moved this to our new GitHub discussion board.

As for your question, does it redirect you to `https`? If not you might need to enter it manually: `https://domain.com:8080`.

**Reply by code-asher** [reply]
If that doesn't do it, it'd help to describe your setup, like how you've set up Let's Encrypt.

**Reply by gelintonx** [reply]
Yes I tried to enter https manually but it didn´t work. I installed lets encrypt with python certbot command.

**Reply by code-asher** [reply]
What's serving the certificates? Django itself or do you have a reverse proxy or something?

**Reply by gelintonx** [reply]
I don´t understand what you mean with the first question. No I haven´t any reverse proxy to django/node servers, my unique reverse proxy is to code-server. I  only access to the port. I insert https://mydomain.com:8000 and it doesn´t work. With HTTP works fine.

**Reply by code-asher** [reply]
For the first question, something needs to be using the certificates in order to provide HTTPS to the browser, if that makes sense. 

For example code-server has the `--cert` and `--cert-key` flags. If you give it the paths to your Let's Encrypt certificates then it will be able to serve HTTPS. It won't affect django at all though.

**Comment by code-asher** [comment, marked as answer]
I'll drop this link here in case it's useful: https://github.com/cdr/code-server/blob/master/doc/guide.md#lets-encrypt

**Reply by gelintonx** [reply]
I used this command to intall let´s encrypt : sudo apt install -y nginx certbot python-certbot-nginx and this other to activate it:  sudo certbot --non-interactive --redirect --agree-tos --nginx -d mydomain.com -m me@example.com.  It´s there any thing wrong?

**Reply by code-asher** [reply]
Ahhh OK then instead of accessing ports directly you'll need to go through nginx so it can handle the TLS (we recommend Caddy though since it can be a little easier to set up and does Let's Encrypt automatically). 

You'll need a configuration like this: https://github.com/cdr/code-server/blob/master/doc/guide.md#nginx That example doesn't show how to set up TLS though, check out https://nginx.org/en/docs/http/configuring_https_servers.html for an example of that.

Once that's set up you can configure it in any way you'd like. For example you could use sub-paths like `mydomain.com/code-server` and `mydomain.com/django` and use `proxy_pass` to go to ports 8080 and 8000. Or you could use sub-domains like `code-server.mydomain.com` and `django.mydomain.com` (you'll need to re-run certbot with those sub-domains for that to work though).

**Reply by gelintonx** [reply]
Ok I will try that. Thank you a lot

**Reply by nhooyr** [reply]
> That example doesn't show how to set up TLS though

It does!

**Reply by gelintonx** [reply]
All is working fine now. Thank you so much. Greetings

**Reply by code-asher** [reply]
@gelintonx Awesome!

@nhooyr It shows setting up TLS with nginx?

**Reply by nhooyr** [reply]
> @nhooyr It shows setting up TLS with nginx?

Yea! See step 5.

<img width="1082" alt="Screen Shot 2020-08-26 at 1 48 58 PM" src="https://user-images.githubusercontent.com/10180857/91338417-e6231b00-e7a2-11ea-8329-e048281971fb.png">

**Reply by code-asher** [reply]
Ohhhh I see! `certbot` must be doing something to the configuration then? So you don't have to manually add the `listen 443 ssl;` and all that to your config.

**Reply by code-asher** [reply]
Yup checked the docs and it says it edits your configuration. Neat, I stand corrected!
